internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -977,8 +1003,11 @@ func flattenAdvancedReplicationSpec(ctx context.Context, apiObject *admin.Replic
 	}
 
 	tfMap := map[string]any{}
-	tfMap["id"] = apiObject.GetId()
 	tfMap["external_id"] = apiObject.GetId()
+
+	// define num_shards for backwards compatibility as this attribute has default value of 1.
+	tfMap["num_shards"] = 1
```

Comment:
now that we are deciding to add deprecation message, do we still want to do this? or should we wait?

Reply:
Independent if we deprecate or not, I believe we still have to define this value to avoid non-empty plans: the schema defines a default value of 1, meaning existing state has a value of 1 and if value is not set in the read a diff is detected in the optional attribute.